diff --git a/d-i/source/partman-auto/recipes-amd64-efi/atomic b/d-i/source/partman-auto/recipes-amd64-efi/atomic
index 927ace0..77ef55f 100644
--- a/d-i/source/partman-auto/recipes-amd64-efi/atomic
+++ b/d-i/source/partman-auto/recipes-amd64-efi/atomic
@@ -22,10 +22,3 @@ partman-auto/text/atomic_scheme ::
 	use_filesystem{ }
 	$default_filesystem{ }
 	mountpoint{ / } .
-
-100% 512 200% linux-swap
-	$defaultignore{ }
-	$lvmok{ }
-	$reusemethod{ }
-	method{ swap }
-	format{ } .
